agent-to-agent token exchange (#6220)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/ea-ai-agent-token-exchange/main/agent-to-agent/exchange-token-for-idjag1.md

> **Note**: This request example uses the access token subject token type.

```bash
  curl --location --request POST \
    --url 'https://{yourOktaDomain}/oauth2/v1/token' \
    --header "Content-Type: application/x-www-form-urlencoded" \
    --header "Accept: application/json" \
    --data-urlencode "grant_type=urn:ietf:params:oauth:grant-type:token-exchange" \
    --data-urlencode "subject_token=eyJraWQiOiJQLVgxeC1ITWtuSThPS0lUeE5TWVlsMHR0bl...." \
    --data-urlencode "subject_token_type=urn:ietf:params:oauth:token-type:access_token" \
    --data-urlencode "requested_token_type=urn:ietf:params:oauth:token-type:id-jag" \
    --data-urlencode "audience=https://{yourOktaDomain}/oauth2/{authServerId}" \
    --data-urlencode "resource=https://agent2.example.com" \
    --data-urlencode "scope=chat.read+chat.history" \
    --data-urlencode "client_assertion_type=urn:ietf:params:oauth:client-assertion-type:jwt-bearer" \
    --data-urlencode "client_assertion=eyJhbGciOiJSUzI1NiIsInR5…[jwt]"
```

| Parameter | Description and value |
| --- | --- |
| `grant_type` | Standard OAuth 2.0 token exchange grant. The value must be `urn:ietf:params:oauth:grant-type:token-exchange`. |

> | `subject_token` | A valid ID or access token that satisfies a delegation link for the AI agent |

Reviewer: ```suggestion
| `subject_token` | A valid ID or access token that satisfies a delegation link for the AI agent. |
```

| `subject_token_type` | The type of subject token. The value is either `urn:ietf:params:oauth:token-type:id_token` or `urn:ietf:params:oauth:token-type:access_token`. |
| `requested_token_type` | The type of token being requested. The value must be `urn:ietf:params:oauth:token-type:id-jag`. |

> | `audience` | The issuer URL of the resource app's authorization server |

Reviewer: ```suggestion
| `audience` | The issuer URL of the resource app's authorization server. |
```

> | `resource` | The resource URL of Agent 2 |

Reviewer: ```suggestion
| `resource` | The resource URL of Agent 2. |
```

> | `resource` | The resource URL of the agent that receives and validates the incoming request |

Reviewer 2: ```suggestion
| `resource` | The resource URL of Agent 2 |
```

> | `scope`    | A list of scopes at the resource app being requested. This defines the permissions for the final access token. |

Reviewer: ```suggestion
| `scope`    | A list of scopes at the resource app that's being requested. This defines the permissions for the final access token. |
```

> | `client_assertion_type` | The type of assertion for client authentication. The value must be `urn:ietf:params:oauth:client-assertion-type:jwt-bearer` |

Reviewer: ```suggestion
| `client_assertion_type` | The type of assertion for client authentication. The value must be `urn:ietf:params:oauth:client-assertion-type:jwt-bearer`. |
```

> | `client_assertion_type` | The type of assertioin. The value must be `urn:ietf:params:oauth:client-assertion-type:jwt-bearer` |

Reviewer 2: ```suggestion
| `client_assertion_type` | The type of assertion for client authentication. The value must be `urn:ietf:params:oauth:client-assertion-type:jwt-bearer` |
```
